Runtime/JSValue.cs

```diff
@@ -10,31 +10,29 @@ namespace NodeApi;
 public readonly struct JSValue : IEquatable<JSValue>
 {
     private readonly napi_value _handle;
-    private readonly JSValueScope? _scope;
-
-    public readonly JSValueScope Scope =>
-        _scope ?? JSValueScope.Current ?? throw new InvalidOperationException("No current scope");
+    private readonly napi_env _scope;
```

Comment:
>private readonly napi_env _scope; [](http://example.com/codeflow?start=4&length=33)

We effectively trading performance for safety. While it may seem a good approach, it goes against our ability using this framework in server environment.

I do not like this change. After that any programming error will result in memory corruption which will be every expensive or impossible to find. I spent too many weeks catching memory corruptions.

If we want speed, then we must find an alternative, but do not trade safety of the code.

Reply:
I agree memory safety is important.

I think I was confusing the `napi_env` and `napi_handle_scope` in this change (because they're sort of combined now in `JSValueScope`) and assuming a closed scope would result in a `napi_invalid_arg` status from any calls using handles from that scope. But the `napi_env` is still valid, and if the `napi_value` is literally just a pointer, then it could be dereferenced after its scope was closed and memory was freed.

I'll try to find a better solution.

Also we'll need some tests for handle scopes including negative tests. I see there are [some node-addon-api handlescope tests](https://github.com/nodejs/node-addon-api/blob/main/test/handlescope.cc) that could be ported over, though it doesn't look like that includes negative tests.